subsys/net/lib/lwm2m/lwm2m_rw_oma_tlv.c

```diff
@@ -537,48 +526,63 @@ static size_t put_float(struct lwm2m_output_context *out,
 	return len;
 }
 
-static size_t put_bool(struct lwm2m_output_context *out,
-		       struct lwm2m_obj_path *path, bool value)
+static int put_bool(struct lwm2m_output_context *out,
+		    struct lwm2m_obj_path *path, bool value)
 {
 	int8_t value_s8 = (value != 0 ? 1 : 0);
 
 	return put_s8(out, path, value_s8);
 }
 
-static size_t put_opaque(struct lwm2m_output_context *out,
-			 struct lwm2m_obj_path *path,
-			 char *buf, size_t buflen)
+static int put_opaque(struct lwm2m_output_context *out,
+		      struct lwm2m_obj_path *path, char *buf, size_t buflen)
 {
 	return put_string(out, path, buf, buflen);
 }
 
-static size_t put_objlnk(struct lwm2m_output_context *out,
-			 struct lwm2m_obj_path *path,
-			 struct lwm2m_objlnk *value)
+static int put_objlnk(struct lwm2m_output_context *out,
+		      struct lwm2m_obj_path *path, struct lwm2m_objlnk *value)
 {
-	int32_t value_s32 = (value->obj_id << 16) | value->obj_inst;
+	struct tlv_out_formatter_data *fd;
+	struct oma_tlv tlv;
+	int32_t net_value = sys_cpu_to_be32(
+				((value->obj_id) << 16) | value->obj_inst);
 
-	return put_s32(out, path, value_s32);
+	fd = engine_get_out_user_data(out);
+	if (!fd) {
+		return -EINVAL;
+	}
+
+	tlv_setup(&tlv, tlv_calc_type(fd->writer_flags),
+		  tlv_calc_id(fd->writer_flags, path), sizeof(net_value));
+
+	return oma_tlv_put(&tlv, out, (uint8_t *)&net_value, false);
 }
 
-static size_t get_number(struct lwm2m_input_context *in, int64_t *value,
-			 uint8_t max_len)
+static int get_number(struct lwm2m_input_context *in, int64_t *value,
+		      uint8_t max_len)
 {
 	struct oma_tlv tlv;
-	size_t size = oma_tlv_get(&tlv, in, false);
+	int size;
 	int64_t temp;
+	int ret;
+
+	size = oma_tlv_get(&tlv, in, false);
+	if (size < 0) {
+		return size;
+	}
```

Comment:
I prefer this early return, because that makes it clear what we do for errors. But we don't use early return in the other get functions, so maybe convert all to early returns?

Reply:
I've updated other functions to return early as well, instead of nesting code within if.